src/main/java/org/codehaus/mojo/aspectj/AbstractAjcMojo.java

```diff
@@ -69,6 +69,15 @@
     @Parameter
     protected String[] weaveDirectories;
 
+    /**
+     * Java 9+ modules to build the module path from.
+     * Corresponds to <code>ajc --module-path</code> option.
+     *
+     * @since 1.13
+     */
+    @Parameter
+    protected Module[] javaModules;
```

Comment:
As you might have noticed, the maven-compiler-plugin doesn't need this to figure out the module path. Please look at that code combined with https://codehaus-plexus.github.io/plexus-languages/plexus-java/usage.html how to solve this without any extra parameters.

Reply:
I do not use modules, I added this parameter a couple of months ago because there was a use case in which modules simply did not work AspectJ Maven, or at least I could not figure out how. So I exposed this parameter to the user, and he could solve his problem. I do not remember the particulars, but I can certainly find the information somewhere in my inbox. If you want to review that case and explain to the user, how he could have done without it, feel free to give it a shot. I am certainly not the most competent person to do so. One thing I do know, though, because I had a discussion about it with Andy Clement, the AspectJ project lead: AspectJ is not particularly well prepared for JMS, because the concept of AOP and cross-cutting concerns is somewhat orthogonal to it. So whatever works, we are happy about it. Of course evenything should work just like with AJC or ECJ, as long as AJC is used as a drop-in replacement for those. But as soon as aspects are involved, JMS is not so much fun anymore, because user expectations about how and what might work, might be different from reality. Sorry to digress, but it is a wider topic.

What I do remember is: A user used this parameter to solve a problem, which is why I added it. Again, we should not patronise users. To be able to put a dependency on the module path sounds like quiote the reasonable feature in a JMS situation.

Reply:
I was a member of the Expert Group and solved the module path issues for the Maven plugins.

Reply:
I am aware of your senior status regarding Maven. But last time I checked, this was not Maven Compiler Plugin, i.e. it does not have the same capabilities and I am certainly not competent to port them, if that is what you are suggesting. So my idea for now is that we add this parameter, however imperfect the solution is from your perspective, until someone comes along and implements it here in the same way Maven Compiler does. I agree that would be better, but for now having this option is better than nothing. If you would volunteer to replace it with a better PR doing what you suggest, I would be glad to give up this option. But for now, I am not. WDYT? Would you volunteer to support us? That would be great and much better than just criticising me for exposing this compiler option to users in need of some JMS support.

Related question: I lately looked into the existing, but unfinished AspectJ component in Plexus Compiler and created https://github.com/codehaus-plexus/plexus-compiler/pull/137 in order to at least fix basic functionality. Would I assume correctly that using Maven Compiler via Plexus in order to compile AspectJ would automatically address this issue?

Reply:
Hi Alex, is your offer to look for that mail in your inbox still a thing? I wonder if there is a better solution for the user you mentioned.
I wonder if https://github.com/codehaus-plexus/plexus-compiler/pull/137 fixed this problemm too.
Have you tried it?

Reply:
> is your offer to look for that mail in your inbox still a thing?

Actually, it was [StackOverflow #65286736](https://stackoverflow.com/q/65286736/1082681). In my inbox was simply the RSS feed notifiying me of new AspectJ questions there. Please read both the question and the answer, if you want to know in more detail. I had a longer private e-mail exchange about this and related topics with Andy Clement, but - it was private. In public, related discussion was in [this thread](https://www.eclipse.org/lists/aspectj-users/msg15493.html) on the AspectJ users mailing list.

> I wonder if codehaus-plexus/plexus-compiler#137 fixed this problem too.

That commit was completely unrelated to this issue or JMS in general. It would not fix that issue, but rather profit from it to be a non-issue in Maven Compiler, because there according to Robert it is solved in another (better) way. That would not help us here, though, like I said - unless someone would port the same functionality to AspectJ Maven. Actually I did not try if this would automatically work in Plexus Compiler, because I am still waiting for Robert to answer this very question from my previous comment here, because he is the expert on the subject matter. @rfscholte, any insigths for Ben and me here? Sorry for repeatedly mentioning you directly. You can simply answer whenever it is the most convenient time for you, I know you must be quite busy.

Reply:
Have you read https://codehaus-plexus.github.io/plexus-languages/plexus-java/usage.html ? Basically: input is `module-info` + jars, output are the jars divided over modulepath and classpath. 
Looking at the SO issue this should really be straight forward.